python_modules/dagster/dagster_tests/components_tests/resolution_tests/test_resolved_from.py

```diff
@@ -99,3 +100,113 @@ def build_defs(self, _):  # type: ignore
     )
     assert c.thing.foo
     assert c.num == 123
+
+
+def test_union_resolvable():
+    class FooModel(Model):
+        foo: str
+
+    class BarModel(Model):
+        bar: str
+
+    @dataclass
+    class ResolveFromListComponent(Component, Resolvable):
+        thing: Union[FooModel, BarModel]
+
+        def build_defs(self, _): ...
+
+    c = load_component_for_test(
+        ResolveFromListComponent,
```

Comment:
we also have `ResolveFromListComponent.resolve_from_...` methods which i think could probably replace the need for `load_component_for_test`